### PR TITLE
Update to controller-tools v0.2.0-beta.5

### LINKED ISSUE #68: Update controller-tools to v2 beta5
/kind feature

**Describe the solution you'd like**
Update controller-tools to v2 beta5. See https://github.com/kubernetes-sigs/cluster-api/pull/1224 as an example.

### PR DESCRIPTION
**Pull request by ncdc**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #68

**Special notes for your reviewer**:
This also adds a new hack/tools go mod.

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -85,4 +85,4 @@ docker-push:
 # Marking this as PHONY so it will build every time (so you always have the correct version locally)
 .PHONY: $(CONTROLLER_GEN)
 $(CONTROLLER_GEN):
-	go build -o $(CONTROLLER_GEN) sigs.k8s.io/controller-tools/cmd/controller-gen
+	cd hack/tools && go build -o $(CONTROLLER_GEN) sigs.k8s.io/controller-tools/cmd/controller-gen
```
Comment (detiber) [inline Makefile]: Wouldn't this give us a potential diff between the vendored version of controller-tools for project and for the controller generation here?

Comment (chuckha) [inline Makefile]: bin ends up in hack/tools/, not top level directory. I ended up make not able to find `bin/controller-gen`

Comment (chuckha) [reply Makefile]: I would recommend this change:

```
CONTROLLER_GEN := hack/tools/bin/controller-gen
```

Comment (chuckha) [reply Makefile]: wait no this is completely wrong

Comment (chuckha) [reply Makefile]: Updated recommendation:

```
TOOLS_DIR := hack/tools
CONTROLLER_GEN := bin/controller-gen
CONTROLLER_GEN_PATH := $(TOOLS_DIR)/$(CONTROLLER_GEN)
```

```
# Build controller-gen
$(CONTROLLER_GEN_PATH): $(TOOLS_DIR)/go.mod
	cd $(TOOLS_DIR) && go build -o $(CONTROLLER_GEN) sigs.k8s.io/controller-tools/cmd/controller-gen
```

I'm kind of making stuff up here with removing the .PHONY but idk, might be cool?

Also changing all the CONTROLLER_GEN references to CONROLLER_GEN_PATH

Comment (ncdc) [reply Makefile]: Updated with a slight variation on your suggestion - thanks @chuckha. PTAL!

File: Makefile [hunk 2]
```diff
@@ -82,7 +84,5 @@ docker-push:
 	docker push ${MANAGER_IMAGE}
 
 # Build controller-gen
-# Marking this as PHONY so it will build every time (so you always have the correct version locally)
-.PHONY: $(CONTROLLER_GEN)
-$(CONTROLLER_GEN):
-	go build -o $(CONTROLLER_GEN) sigs.k8s.io/controller-tools/cmd/controller-gen
+$(CONTROLLER_GEN): $(TOOLS_DIR)/go.mod
```
Comment (detiber) [inline Makefile]: I still worry that we have potentially reintroduced drift here between the version of controller-gen we are using here and the version of controller-tools we are vendoring for the controller.

Comment (BenTheElder) [reply Makefile]: hmm, maybe pin both with a replace via some module tidy script?

Comment (ncdc) [reply Makefile]: @detiber it's not vendored anywhere else. It's only used here. I need to regen the main go.sum to make it go away.

Comment (ncdc) [reply Makefile]: Updated - ptal

**Comment by ncdc** [comment]
Yay CI passed 💚

File: Makefile (hunk 2 above)
Comment (chuckha) [reply Makefile]: I think the concern is that controller-runtime needs to match controller-tools?

**Comment by chuckha** [comment]
Looks good to me, I don't know enough about the version dependency between `controller-runtime` and `controller-tools`

Do those have to match?

**Comment by detiber** [comment]
> Looks good to me, I don't know enough about the version dependency between controller-runtime and controller-tools

> Do those have to match?

Not necessarily, but they should be kept compatible. That is outside the scope of this PR, though :)

**Comment by ncdc** [comment]
I'm not sure how strict the relationship is between controller-tools and controller-runtime. Right now, for example. controller-tools is at beta.5 while controller-runtime is still at beta.4.

**Comment by chuckha** [comment]
In that case, I think this is good to go

/approve

**Pull request merged**